cmake/precompileheaders.cmake

```diff
@@ -27,118 +27,55 @@
-            set(ignorePaths "")
-        endif()
-        list(APPEND ignorePaths
-            ${publicIgnorePaths}
-            "${CMAKE_CURRENT_SOURCE_DIR}"
-            "${CMAKE_CURRENT_BINARY_DIR}"
-            "${IVW_EXTENSIONS_DIR}/warn"
-        )
-        if(WIN32 AND MSVC)
-            ivw_get_drive(${ivw_private_windows_path} windrive)
-            list(APPEND ignorePaths ${windrive})
-        endif()
-        list(REMOVE_DUPLICATES ignorePaths)
-
-        ivw_get_target_property_recursive(publicIncludePaths ${target} COTIRE_PREFIX_HEADER_PUBLIC_INCLUDE_PATH False)
-        get_target_property(includePaths ${target} COTIRE_PREFIX_HEADER_INCLUDE_PATH)
-        if(NOT includePaths)
-            set(includePaths "")
-        endif()
-        list(APPEND includePaths
-            "${IVW_ROOT_DIR}"
-            ${publicIncludePaths}
-        )
-        list(REMOVE_DUPLICATES includePaths)
+    ivw_target_pch_headers(${target} REUSE)
+endfunction()
 
-        set_target_properties(${target} PROPERTIES COTIRE_PREFIX_HEADER_IGNORE_PATH "${ignorePaths}")
-        set_target_properties(${target} PROPERTIES COTIRE_PREFIX_HEADER_INCLUDE_PATH "${includePaths}")
-        set_target_properties(${target} PROPERTIES COTIRE_ADD_UNITY_BUILD FALSE)
+# Set which headers files should be used for pre-compiled headers of given target.
+# If the REUSE option is used, target will reuse the PCH from inviwo::core.
+function(ivw_target_pch_headers target)
+    set(options "REUSE")
+    set(oneValueArgs "VISIBILITY")
+    ## VISIBILITY should be INTERFACE;PUBLIC;PRIVATE (default)
+    set(multiValueArgs "HEADERS")
+    cmake_parse_arguments(PARSE_ARGV 1 ARG ${options} ${oneValueArgs} ${multiValueArgs})
+
+    if(NOT DEFINED ARG_VISIBILITY)
+        set(ARG_VISIBILITY "PRIVATE")
+    endif()
 
-        cotire(${target})
+    if(IVW_CFG_PRECOMPILED_HEADERS AND (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16.0"))
+        if(ARG_REUSE)
+            ivw_get_target_depending_on(dependencies ${target} TRUE)
+            if ("${dependencies}" MATCHES "inviwo::core")
+                # ensure the same compiler definitions
+                ivw_define_standard_properties(${target})
+                target_precompile_headers(${target} REUSE_FROM inviwo-core-pch)
+            endif()
+        else()
+            target_precompile_headers(${target} ${ARG_VISIBILITY} ${ARG_HEADERS})
+            file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/pch-dummy.cpp 
+                "// Dummy source file needed in target for reusing pch artifacts\n"
+            )
+            add_library(${target}-pch STATIC ${CMAKE_CURRENT_BINARY_DIR}/pch-dummy.cpp)
+            ivw_define_standard_definitions(${target}-pch ${target}-pch)
+            ivw_define_standard_properties(${target}-pch)
+            target_link_libraries(${target}-pch PRIVATE ${target})
+            target_precompile_headers(${target}-pch ${ARG_VISIBILITY} ${ARG_HEADERS})
+            ivw_folder(${target}-pch "pch-targets")
```

Comment:
What is all this about?

Reply:
See https://discourse.cmake.org/t/help-with-target-precompile-headers/1065 why a dummy target is needed.

Reply:
that is some shady stuff, and seems mostly concerned with static linking and object libraries, neither which I think you are using here. So please elaborate on why this is needed?